readthedocs/proxito/views/serve.py

```diff
@@ -303,9 +303,12 @@ def get(self, request, proxito_path, template_name='404.html'):
         # ``redirect_filename`` is the path without ``/<lang>/<version>`` and
         # without query, starting with a ``/``. This matches our old logic:
         # https://github.com/readthedocs/readthedocs.org/blob/4b09c7a0ab45cd894c3373f7f07bad7161e4b223/readthedocs/redirects/utils.py#L60
-        # We parse ``filename`` to remove the query from it
-        schema, netloc, path, params, query, fragments = urlparse(filename)
-        redirect_filename = path
+        # We parse ``filename`` to remove the query from it,
+        # and to remove invalid chars from it.
+        # We don't use ``.path`` to avoid parsing the filename as
+        # a full url.
+        parsed = urlparse(filename)
+        redirect_filename = parsed._replace(query="").geturl()
```

Comment:
I'm not even sure that the filename has any query strings at this point.

Reply:
I'm not sure to understand this change. Can you expand a little more here? What would be a valid `filename` here and what would be the resultant `redirect_filename` value?

Reply:
Cool! One thing that still confuses me here is that it's the variable is called `filename` but we are saying that it could be full URL (e.g. `http://example.com/my-path`). Is that because the original URL could be `https://mydocs.com/\rhttp://example.com/my-path`? Is that the case?

Reply:
@humitos yeah, that's the case where we can end up with that filename.